### PR TITLE
refactor(frontend): intergrate useErrorSummary hook in protected routes

### PR DESCRIPTION
**Pull request by sebastien-comeau**
### Description

As per title.

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/routes/$lang/_protected/personal-information/mailing-address/edit.tsx [hunk 1]
```diff
@@ -192,43 +195,25 @@ export async function action({ context: { session }, params, request }: ActionFu
...
-  }, [errorMessages]);
+  const errors = fetcher.data?.errors;
+  const errorSummary = useErrorSummary(errors, {
+    streetName: 'street-name',
+    apartment: 'apartment-number',
+    countryId: 'country-id',
+    provinceTerritoryStateId: 'province-id',
+    cityName: 'city-name',
+    postalCode: 'postal-code',
+    homeAndMailingAddressTheSame: '',
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang/_protected/personal-information/mailing-address/edit.tsx:215]:
```suggestion
    homeAndMailingAddressTheSame: 'input-checkbox-homeAndMailingAddressTheSame',
```

File: frontend/app/routes/$lang/_protected/personal-information/home-address/edit.tsx [hunk 2]
```diff
@@ -197,42 +200,24 @@ export async function action({ context: { session }, params, request }: ActionFu
...
-  }, [errorMessages]);
+  const errors = fetcher.data?.errors;
+  const errorSummary = useErrorSummary(errors, {
+    streetName: 'street-name',
+    apartment: 'apartment-number',
+    countryId: 'country-id',
+    provinceTerritoryStateId: 'province-id',
+    cityName: 'city-name',
+    postalCode: 'postal-code',
+    homeAndMailingAddressTheSame: '',
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang/_protected/personal-information/home-address/edit.tsx:219]:
```suggestion
    homeAndMailingAddressTheSame: 'input-checkbox-homeAndMailingAddressTheSame',
```

File: frontend/app/routes/$lang/_protected/personal-information/mailing-address/edit.tsx (hunk 1 above)
Comment (sebastien-comeau) [reply frontend/app/routes/$lang/_protected/personal-information/mailing-address/edit.tsx]: Fixed

File: frontend/app/routes/$lang/_protected/personal-information/home-address/edit.tsx (hunk 2 above)
Comment (sebastien-comeau) [reply frontend/app/routes/$lang/_protected/personal-information/home-address/edit.tsx]: Fixed

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**